eventsourcing/src/main/java/org/axonframework/eventsourcing/AbstractSnapshotter.java

```diff
@@ -217,6 +236,17 @@ public Builder executor(Executor executor) {
             return this;
         }
 
+        /**
+         * Sets the {@link SpanFactory} implementation to use for providing tracing capabilities.
+         *
+         * @param spanFactory The {@link SpanFactory} implementation
+         * @return The current Builder instance, for fluent interfacing.
+         */
+        public Builder spanFactory(SpanFactory spanFactory) {
+            this.builderSpanFactory = spanFactory;
```

Comment:
I'd expect the `SpanFactory` to be validated for not being `null` to protect the users during start-up.